sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs

```diff
@@ -33,4 +42,47 @@ public ModelReaderWriterOptions (string format)
     /// Gets the format to read and write the model.
     /// </summary>
     public string Format { get; }
+
+    /// <summary>
+    /// .
+    /// </summary>
+    public void AddProxy<T>(Type type, IPersistableModel<T> proxy)
+    {
+        if (_proxies is null)
+        {
+            _proxies = new Dictionary<Type, object>();
+        }
+
+        //multiple of same type?
+        _proxies.Add(type, proxy);
+    }
+
+    /// <summary>
+    /// .
+    /// </summary>
+    public object? WriteContext { get; private set; }
```

Comment:
I thought about it more and I think we need to be more explicit what's stored here. I also want to tie the name to the AddProxy method. What about calling it ProxiedModel or ProxyModel?